File: Runtime/Plugins/Android/ECommerceEventSerializer.java
package io.appmetrica.analytics.plugin.unity;

import androidx.annotation.NonNull;
import org.json.JSONException;
import org.json.JSONObject;
import io.appmetrica.analytics.ecommerce.ECommerceEvent;

final class ECommerceEventSerializer {
    private ECommerceEventSerializer() {}

    @NonNull
    public static ECommerceEvent fromJsonString(@NonNull String ecommerce) throws JSONException {
        JSONObject json = new JSONObject(ecommerce);
        String type = json.getString("Type");
        switch (type) {
            case "AddCartItemECommerceEvent":
                return ECommerceEvent.addCartItemEvent(
                    ECommerceCartItemSerializer.fromJsonString(json.getString("CartItem"))
                );
            case "BeginCheckoutECommerceEvent":
                return ECommerceEvent.beginCheckoutEvent(
                    ECommerceOrderSerializer.fromJsonString(json.getString("Order"))
                );
            case "PurchaseECommerceEvent":
                return ECommerceEvent.purchaseEvent(
                    ECommerceOrderSerializer.fromJsonString(json.getString("Order"))
                );
            case "RemoveCartItemECommerceEvent":
                return ECommerceEvent.removeCartItemEvent(
                    ECommerceCartItemSerializer.fromJsonString(json.getString("CartItem"))
                );
            case "ShowProductCardECommerceEvent":
                return ECommerceEvent.showProductCardEvent(
                    ECommerceProductSerializer.fromJsonString(json.getString("Product")),
                    ECommerceScreenSerializer.fromJsonString(json.getString("Screen"))
                );
            case "ShowProductDetailsECommerceEvent":
                return ECommerceEvent.showProductDetailsEvent(
                    ECommerceProductSerializer.fromJsonString(json.getString("Product")),
                    json.has("Referrer") ? ECommerceReferrerSerializer.fromJsonString(json.getString("Referrer")) : null
                );
            case "ShowScreenECommerceEvent":
                return ECommerceEvent.showScreenEvent(
                    ECommerceScreenSerializer.fromJsonString(json.getString("Screen"))
                );
        }
        throw new JSONException("Unknown ECommerce type " + type);
    }
}

File: Runtime/Native/Utils/Serializer/ECommerceEventSerializer.cs
using System.Collections.Generic;
using Io.AppMetrica.Ecommerce;
using Io.AppMetrica.Internal;
using Io.AppMetrica.Internal.Ecommerce;
using JetBrains.Annotations;

namespace Io.AppMetrica.Native.Utils.Serializer {
    internal static class ECommerceEventSerializer {
        [CanBeNull]
        public static string ToJsonString([NotNull] this ECommerceEvent self) {
            switch (self) {
                case AddCartItemECommerceEvent @event:
                    return JSONEncoder.Encode(new Dictionary<string, object> {
                        { "Type", "AddCartItemECommerceEvent" },
                        { "CartItem", @event.CartItem.ToJsonString() },
                    });
                case BeginCheckoutECommerceEvent @event:
                    return JSONEncoder.Encode(new Dictionary<string, object> {
                        { "Type", "BeginCheckoutECommerceEvent" },
                        { "Order", @event.Order.ToJsonString() },
                    });
                case PurchaseECommerceEvent @event:
                    return JSONEncoder.Encode(new Dictionary<string, object> {
                        { "Type", "PurchaseECommerceEvent" },
                        { "Order", @event.Order.ToJsonString() },
                    });
                case RemoveCartItemECommerceEvent @event:
                    return JSONEncoder.Encode(new Dictionary<string, object> {
                        { "Type", "RemoveCartItemECommerceEvent" },
                        { "CartItem", @event.CartItem.ToJsonString() },
                    });
                case ShowProductCardECommerceEvent @event:
                    return JSONEncoder.Encode(new Dictionary<string, object> {
                        { "Type", "ShowProductCardECommerceEvent" },
                        { "Product", @event.Product.ToJsonString() },
                        { "Screen", @event.Screen.ToJsonString() },
                    });
                case ShowProductDetailsECommerceEvent @event:
                    return JSONEncoder.Encode(new Dictionary<string, object> {
                        { "Type", "ShowProductDetailsECommerceEvent" },
                        { "Product", @event.Product.ToJsonString() },
                        { "Referrer", @event.Referrer?.ToJsonString() },
                    });
                case ShowScreenECommerceEvent @event:
                    return JSONEncoder.Encode(new Dictionary<string, object> {
                        { "Type", "ShowScreenECommerceEvent" },
                        { "Screen", @event.Screen.ToJsonString() },
                    });
                default:
                    return null;
            }
        }
    }
}
